packages/ckit/src/providers/mercury/MercuryProvider.ts

```diff
@@ -24,21 +24,44 @@ function toCell(resolved: ResolvedOutpoint): Cell {
   };
 }
 
+interface BatchRequest {
+  method: string;
+  params: unknown[];
+}
+
 export class MercuryProvider extends AbstractProvider {
   readonly mercury: MercuryClient;
   readonly rpc: RPC;
+  readonly rpcUrl: string;
 
-  constructor(
-    mercuryRpc: MercuryClient | string = 'http://127.0.0.1:8116',
-    ckbRpc: RPC | string = 'http://127.0.0.1:8114',
-  ) {
+  constructor(mercuryRpc = 'http://127.0.0.1:8116', ckbRpc = 'http://127.0.0.1:8114') {
     super();
 
-    if (mercuryRpc instanceof MercuryClient) this.mercury = mercuryRpc;
-    else this.mercury = new MercuryClient(mercuryRpc);
+    this.mercury = new MercuryClient(mercuryRpc);
+    this.rpc = new RPC(ckbRpc);
+    this.rpcUrl = ckbRpc;
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  async batchRequestCkb(request: BatchRequest): Promise<any> {
```

Comment:
- use a generic `<T>` rather than the `any`
- `BatchRequest` -> `BatchRequest[]`